### PR TITLE
Mock tracking

### PR DESCRIPTION
**Pull request by nickpatrick**
https://app.clubhouse.io/radarlabs/story/2605/ios-mock-tracking

```swift
Radar.mockTracking(
  origin: origin,
  destination: destination,
  mode: .car,
  points: 10,
  interval: 2
) { (status, location, events, user) in
  // ...
}
```

### CODE REVIEW

**Comment by nickpatrick** [comment]
@blackmad FYI

File: RadarSDK/RadarAPIClient.h [hunk 1]
```diff
@@ -90,9 +93,15 @@ typedef void (^_Nullable RadarRouteAPICompletionHandler)(RadarStatus status, NSD
 
 - (void)getDistanceFromOrigin:(CLLocation *_Nonnull)origin
                   destination:(CLLocation *_Nonnull)destination
-                        modes:(RadarRouteMode)mode
+                        modes:(RadarRouteMode)modes
```
Comment (pingxia0823) [inline RadarSDK/RadarAPIClient.h:96]: nit:  maybe rename it to `mode:(RadarRouteMode)mode` to be consistent with the naming in the new API

File: RadarSDK/Radar.m [hunk 2]
```diff
@@ -111,6 +114,58 @@ + (void)startTrackingWithOptions:(RadarTrackingOptions *)options {
...
+                                             intervalLimit = 60;
+                                         }
+
+                                         for (int i = 0; i < coordinates.count; i++) {
+                                             RadarCoordinate *coordinate = coordinates[i];
+                                             __block CLLocation *location = [[CLLocation alloc] initWithCoordinate:coordinate.coordinate
+                                                                                                          altitude:-1
+                                                                                                horizontalAccuracy:5
+                                                                                                  verticalAccuracy:-1
+                                                                                                         timestamp:[NSDate new]];
```
Comment (pingxia0823) [inline RadarSDK/Radar.m:149]: nit: do we need to give caller more flexibility to specify these additional params (we could use these values as a preset of course)?

File: RadarSDK/RadarCoordinate.m [hunk 3]
```diff
@@ -9,6 +9,63 @@
...
+        ![coordinatesLatitudeObj isKindOfClass:[NSNumber class]]) {
+        return nil;
+    }
+
+    NSNumber *coordinatesLongitudeNumber = (NSNumber *)coordinatesLongitudeObj;
+    NSNumber *coordinatesLatitudeNumber = (NSNumber *)coordinatesLatitudeObj;
+
+    float coordinatesLongitudeFloat = [coordinatesLongitudeNumber floatValue];
+    float coordinatesLatitudeFloat = [coordinatesLatitudeNumber floatValue];
+
```
Comment (pingxia0823) [inline RadarSDK/RadarCoordinate.m:63]: nit: there is a helper method `[NSDictionary radar_coordinateForKey]` in `RadarCollectionsAddition`, which does a bit more thorough checking on coordinate fields from server response

File: RadarSDK/Radar.m (hunk 2 above)
Comment (nickpatrick) [reply RadarSDK/Radar.m:149]: You mean `altitude`, `horizontalAccuracy`, and `verticalAccuracy`? I could see the server returning `altitude`, but most customers probably won't need it. The iOS simulator always uses `horizontalAccuracy = 5` for locations in the simulator, just following the same convention here.

File: RadarSDK/RadarAPIClient.h (hunk 1 above)
Comment (nickpatrick) [reply RadarSDK/RadarAPIClient.h:96]: Distance API should take `modes` (can return routes for multiple modes, comma-separated), whereas mock API should take `mode` (single mode). The param is called `modes` in @blackmad's mock API PR but we should probably change to `mode` before releasing.

File: RadarSDK/RadarCoordinate.m (hunk 3 above)
Comment (nickpatrick) [reply RadarSDK/RadarCoordinate.m:63]: Good call, will use that here

File: RadarSDK/Radar.h [hunk 4]
```diff
@@ -304,9 +318,27 @@ typedef void (^_Nonnull RadarRouteCompletionHandler)(RadarStatus status, RadarRo
...
+ @param mode The travel mode.
+ @param points The number of mock location updates.
+ @param interval The interval in seconds between each mock location update. A number between 2 and 60.
+
+ @see https://radar.io/documentation/sdk#ios-mock
+ */
++ (void)mockTrackingWithOrigin:(CLLocation *_Nonnull)origin
+                   destination:(CLLocation *_Nonnull)destination
+                          mode:(RadarRouteMode)mode
+                        points:(int)points
```
Comment (jsani-radar) [inline RadarSDK/Radar.h:338]: had the same comment on Android - `numPoints`?

**Pull request merged**